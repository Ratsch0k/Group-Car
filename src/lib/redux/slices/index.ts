import {connectRouter} from 'connected-react-router';
import {History} from 'history';
import authReducer from './auth/authSlice';
import groupReducer from './group/groupSlice';

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
const rootReducers = (history: History) => {
  return {
    group: groupReducer,
    auth: authReducer,
    router: connectRouter(history),
  };
};

export default rootReducers;
