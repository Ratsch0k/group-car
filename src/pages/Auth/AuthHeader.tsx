import React from 'react';
import {
  Container,
  createStyles,
  makeStyles,
  Typography,
} from '@material-ui/core';
import {LargeLogo} from '../../lib/components/Icons/LargeLogo';
import {GroupCarTheme} from '../../lib';

const useStyles = makeStyles((theme: GroupCarTheme) => createStyles({
  root: {
    color: theme.palette.primary.main,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column',
    marginBottom: theme.spacing(2),
    flexGrow: 0,
    flexShrink: 0,
  },
}));


const AuthHeader = (): JSX.Element => {
  const classes = useStyles();

  return (
    <Container className={classes.root}>
      <LargeLogo />
      <Typography variant='h5'>
        <b>My Group Car</b>
      </Typography>
    </Container>
  );
};

export default AuthHeader;
