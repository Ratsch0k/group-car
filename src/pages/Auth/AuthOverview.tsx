import {
  Container,
  Grid,
  Link,
  makeStyles,
  Typography,
} from '@material-ui/core';
import React from 'react';
import {Button} from '../../lib';
import {Trans, useTranslation} from 'react-i18next';

const useStyles = makeStyles({
  buttonContainer: {
    maxWidth: 300,
    width: '100%',
  },
  buttonLabel: {
    fontWeight: 'bold',
  },
});


interface AuthButtonProps {
  children: React.ReactNode;
  onClick: React.MouseEventHandler<HTMLButtonElement>;
}

export const AuthButton = (
  {children, onClick}: AuthButtonProps,
): JSX.Element => {
  const classes = useStyles();

  return (
    <Grid item className={classes.buttonContainer}>
      <Button
        variant='contained'
        fullWidth
        onClick={onClick}
        color='primary'
        classes={{
          label: classes.buttonLabel,
        }}
      >
        {children}
      </Button>
    </Grid>
  );
};

export interface AuthOverviewProps {
  toLogin(): void;
  toSignUp(): void;
}

export const AuthOverview = (props: AuthOverviewProps): JSX.Element => {
  const {toLogin, toSignUp} = props;
  const {t} = useTranslation();

  return (
    <Container maxWidth='xs'>
      <Grid
        container
        direction='column'
        alignItems='center'
        spacing={4}
      >
        <Typography align='center' variant='h6' color='textSecondary'>
          {t('auth.title')}
        </Typography>
        <AuthButton onClick={toLogin}>
          {t('form.login')}
        </AuthButton>
        <AuthButton onClick={toSignUp}>
          {t('form.sign-up')}
        </AuthButton>
        <Typography color='textSecondary'>
          <Trans
            i18nKey='auth.tryDemo'
            components={[
              <Link
                referrerPolicy='no-referrer'
                target='_blank'
                href='https://demo.mygroupcar.de'
                key='demo-link'
              />,
            ]}
          />
        </Typography>
      </Grid>
    </Container>
  );
};

export default AuthOverview;
