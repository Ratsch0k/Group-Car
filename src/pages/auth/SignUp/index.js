export {default as SignUpForm} from './SignUpForm';
export {default as SignUpDialog} from './SignUpDialog';
