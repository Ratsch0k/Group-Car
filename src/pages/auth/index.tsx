export {default} from './Authentication';
export * from './Login';
export * from './SignUp';
