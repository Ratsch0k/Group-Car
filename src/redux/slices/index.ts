import authReducer from './auth/authSlice';
import groupReducer from './group/groupSlice';

export default {
  group: groupReducer,
  auth: authReducer,
};
