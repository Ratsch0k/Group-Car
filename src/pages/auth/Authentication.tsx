import {
  Button,
  Dialog,
  DialogContent,
  DialogTitle,
  Divider,
  IconButton,
  makeStyles,
  createStyles,
  Container,
  Typography,
} from '@material-ui/core';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import React, {useState, useEffect, useContext} from 'react';
import {
  Redirect,
  Route,
  Switch,
  useHistory,
  useRouteMatch,
  useLocation,
} from 'react-router-dom';
import {LoginForm} from './Login';
import {SignUpForm} from './SignUp';
import {useTranslation} from 'react-i18next';
import AuthContext from 'lib/context/auth/authContext';
import CloseIcon from '@material-ui/icons/Close';

type Theme = import('@material-ui/core').Theme;

const useStyle = makeStyles((theme: Theme) =>
  createStyles({
    content: {
      paddingBottom: theme.spacing(3),
    },
    divider: {
      margin: theme.spacing(2),
    },
    dividerContainer: {
      position: 'relative',
    },
    dividerText: {
      paddingLeft: theme.spacing(0.5),
      paddingRight: theme.spacing(0.5),
      margin: 'auto',
      top: '50%',
      right: '50%',
      position: 'absolute',
      transform: 'translate(50%, -50%)',
      background: theme.palette.background.paper,
    },
    closeButton: {
      float: 'right',
    },
  }),
);

const Authentication: React.FC = () => {
  const match = useRouteMatch();
  const history = useHistory();
  const classes = useStyle();
  const {t} = useTranslation();
  const location = useLocation();
  const auth = useContext(AuthContext);

  const [loading, setLoading] = useState<boolean>(false);

  /**
   * Save the url from which the client navigated to this component
   */
  const [origin] = useState<string>('/');

  /**
   * Handles navigation to sign up page
   */
  const handleSignUp = () => {
    history.push(`${match.url}/sign-up`);
  };

  /**
   * Handles navigation to login page
   */
  const handleLogin = () => {
    history.push(`${match.url}/login`);
  };

  /**
   * Handles go back action
   */
  const handleBack = () => {
    history.goBack();
  };

  /**
   * Navigates to the page from which auth was called.
   */
  const onFinished = () => {
    history.replace(origin);
  };

  /**
   * Checks every time the `isLoggedIn` attribute changes
   * if the client is authenticated. If it is go to the origin.
   */
  useEffect(() => {
    if (auth.isLoggedIn) {
      onFinished();
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [auth.isLoggedIn]);

  return (
    <Dialog open={true} fullWidth maxWidth='xs'>
      <DialogTitle>
        <IconButton
          data-testid='go-back'
          edge='start'
          disableRipple
          disabled={loading}
          onClick={handleBack}
          style={{visibility: match.isExact ? 'hidden' : 'visible'}}
        >
          <ArrowBackIcon/>
        </IconButton>
        {
          match.isExact ?
          t('auth.dialog.title.authenticate') :
          t('auth.dialog.title.' +
            location.pathname.replace(match.path + '/', ''))
        }
        <IconButton
          className={classes.closeButton}
          onClick={onFinished}
          disabled={loading}
          data-testid='close-dialog'
        >
          <CloseIcon />
        </IconButton>
      </DialogTitle>
      <DialogContent className={classes.content}>
        <Switch>
          <Route exact path={match.path}>
            <Button
              data-testid='to-login'
              fullWidth
              variant='outlined'
              color='primary'
              onClick={handleLogin}
            >
              {t('form.login')}
            </Button>
            <Container className={classes.dividerContainer}>
              <Divider orientation='horizontal' className={classes.divider}/>
              <Typography component='span' className={classes.dividerText}>
                {t('misc.or')}
              </Typography>
            </Container>
            <Button
              data-testid='to-signup'
              fullWidth
              variant='outlined'
              color='primary'
              onClick={handleSignUp}
            >
              {t('form.sign-up')}
            </Button>
          </Route>
          <Route path={`${match.path}/sign-up`}>
            <SignUpForm
              withSubmit
              onFinished={onFinished}
              setLoading={setLoading}
            />
          </Route>
          <Route path={`${match.path}/login`}>
            <LoginForm
              withSubmit
              onFinished={onFinished}
              setLoading={setLoading}
            />
          </Route>
          <Route path='*'>
            <Redirect to='/auth' />
          </Route>
        </Switch>
      </DialogContent>
    </Dialog>
  );
};

export default Authentication;
