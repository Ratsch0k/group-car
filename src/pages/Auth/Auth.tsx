import React, {useCallback, useEffect, useRef} from 'react';
import {
  alpha,
  Box,
  Container,
  createStyles,
  Grid,
  Link,
  makeStyles,
  Paper,
} from '@material-ui/core';
import {
  GradientBackground,
  GroupCarTheme,
  LoginForm,
  SignUpBody,
} from '../../lib';
import {Route, Switch, useRouteMatch} from 'react-router-dom';
import {useAppDispatch, useShallowAppSelector} from '../../lib/redux/hooks';
import {getLocation, push, replace} from 'connected-react-router';
import AuthOverview from './AuthOverview';
import AuthHeader from './AuthHeader';
import {CSSTransition, SwitchTransition} from 'react-transition-group';
import {getIsLoggedIn} from '../../lib/redux/slices/auth';
import useSnackbar from '../../lib/hooks/useSnackbar';
import {useTranslation} from 'react-i18next';
import {goToModal} from 'lib/redux/slices/modalRouter/modalRouterSlice';

const useStyles = makeStyles((theme: GroupCarTheme) => createStyles({
  root: {
    height: '100%',
    width: '100%',
    display: 'flex',
    alignItems: 'center',
  },
  container: {
    backgroundColor: alpha(theme.palette.background.default, 0.5),
    border: `2px solid ${theme.palette.background.default}`,
    margin: 'auto',
    minHeight: '25rem',
    backdropFilter: 'blur(4px)',
    padding: theme.spacing(2),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'stretch',
    overflow: 'hidden',
    borderRadius: theme.shape.borderRadiusSized.large,
  },
  contentWrapper: {
    flexGrow: 1,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    transition: 'height 500ms',
    position: 'relative',
  },
  paperRounded: {
    borderRadius: theme.shape.borderRadiusSized.large,
  },
  paperRoot: {
    backgroundColor: 'unset',
  },
  content: {
    padding: theme.spacing(2),
    paddingBottom: 0,
  },
  contentRoot: {
    zIndex: 10,
  },
  divider: {
    backgroundColor: theme.palette.background.default,
    height: 2,
  },
  enter: {
    opacity: 0,
  },
  enterActive: {
    opacity: 1,
    transition: 'opacity 125ms',
  },
  exit: {
    opacity: 1,
  },
  exitActive: {
    opacity: 0,
    transition: 'opacity 125ms',
  },
  footer: {
    position: 'absolute',
    bottom: 0,
    margin: 'auto',
    width: '100%',
    zIndex: 0,
  },
  footerLinks: {
    margin: 'auto',
    marginBottom: theme.spacing(1),
    padding: theme.spacing(1),
    background: alpha(theme.palette.background.paper, 0.4),
    borderRadius: theme.shape.borderRadiusSized.default,
    width: 'fit-content',
    color: theme.palette.primary.dark,
  },
}));

export interface AuthenticationProps {
  preloadMain(): Promise<unknown>;
}

export const Authentication = (props: AuthenticationProps): JSX.Element => {
  const {preloadMain} = props;
  const classes = useStyles();
  const dispatch = useAppDispatch();
  const contentRef = useRef<HTMLDivElement | null>(null);
  const contentObserver = useRef<ResizeObserver>();
  const contentWrapperRef = useRef<HTMLDivElement | null>(null);
  const location = useShallowAppSelector(getLocation);
  const isLoggedIn = useShallowAppSelector(getIsLoggedIn);
  const {show} = useSnackbar();
  const {t} = useTranslation();
  const {path: basePath} = useRouteMatch();

  /**
   * Handles navigating to the login route
   */
  const handleToLogin = useCallback(() => {
    dispatch(push(`${basePath}/login`));
  }, []);

  /**
   * Handles navigating to the sign-up route
   */
  const handleToSignUp = useCallback(() => {
    dispatch(push(`${basePath}/sign-up`));
  }, []);

  /**
   * Handles navigating back
   */
  const handleGoBack = useCallback(() => {
    dispatch(replace(basePath));
  }, []);

  /**
   * Handles a successful login/sign-up event.
   * Notifies the user that they are authenticated via
   * a snack and preloads the main page. Then it
   * navigates to the root path.
   */
  const handleFinished = useCallback(async () => {
    show('success', t('auth.success'));
    await preloadMain();
    dispatch(push('/'));
  }, []);


  /**
   * Handles the case when this route is accessed while
   * the user is already logged-in. Replaces this route
   * with the root route.
   */
  useEffect(() => {
    const f = async () => {
      if (isLoggedIn) {
        await preloadMain();
        dispatch(replace('/'));
      }
    };
    f();
  }, [isLoggedIn]);

  /*
   * Animate resizing the component
   */
  useEffect(() => {
    if (contentRef.current) {
      const observer = new ResizeObserver(handleResize);
      observer.observe(contentRef.current);

      contentObserver.current = observer;
    }
  }, []);

  /**
   * Handles resize events.
   * Gets the new height of the component and sets the height
   * of the content wrapper to the correct height
   */
  const handleResize: ResizeObserverCallback = useCallback((
    [observerEntry],
  ) => {
    if (contentWrapperRef.current) {
      const height = observerEntry.contentRect.height;
      if (height > 0) {
        contentWrapperRef.current.style.height = height + 'px';
      }
    }
  }, []);

  return (
    <Box className={classes.root}>
      <GradientBackground />
      <Container maxWidth='sm' className={classes.contentRoot}>
        <Paper
          elevation={15}
          classes={{
            rounded: classes.paperRounded,
            root: classes.paperRoot,
          }}
        >
          <Box className={classes.container}>
            <AuthHeader />
            <div className={classes.contentWrapper} ref={contentWrapperRef}>
              <div ref={contentRef} className={classes.content}>
                <SwitchTransition>
                  <CSSTransition
                    key={location.key}
                    timeout={250}
                    classNames={{
                      enter: classes.enter,
                      enterActive: classes.enterActive,
                      exit: classes.exit,
                      exitActive: classes.exitActive,
                    }}
                  >
                    <Switch location={location}>
                      <Route key={basePath} exact path={basePath}>
                        <AuthOverview
                          toLogin={handleToLogin}
                          toSignUp={handleToSignUp}
                        />
                      </Route>
                      <Route
                        key={`${basePath}/login`}
                        path={`${basePath}/login`}
                      >
                        <LoginForm
                          withSubmit
                          withGoBack
                          goBack={handleGoBack}
                          onFinished={handleFinished}
                        />
                      </Route>
                      <Route
                        key={`${basePath}/sign-up`}
                        path={`${basePath}/sign-up`}
                      >
                        <SignUpBody
                          withSubmit
                          goBack={handleGoBack}
                          onFinished={handleFinished}
                        />
                      </Route>
                    </Switch>
                  </CSSTransition>
                </SwitchTransition>
              </div>
            </div>
          </Box>
        </Paper>
      </Container>
      <Box className={classes.footer}>
        <Grid container spacing={2} className={classes.footerLinks}>
          <Grid item>
            <Link
              color='inherit'
              component='button'
              onClick={() => dispatch(goToModal('/privacy-policy'))}
            >
              {t('privacyPolicy.title')}
            </Link>
          </Grid>
          <Grid item>
            <Link
              color='inherit'
              component='button'
              onClick={() => dispatch(goToModal('/imprint'))}
            >
              {t('imprint.title')}
            </Link>
          </Grid>
        </Grid>
      </Box>
    </Box>
  );
};

export default Authentication;
