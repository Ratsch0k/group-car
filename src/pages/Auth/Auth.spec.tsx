import _testRender from "../../__test__/testRender";
import React from 'react';
import {AuthState} from "../../lib/redux/slices/auth";
import Auth from "./Auth";
import {Route} from "react-router-dom";
import history from "../../lib/redux/history";
import {config} from 'react-transition-group';

/**
 * !IMPORTANT
 * Unfortunately it was not possible for me to add more tests which
 * test navigating between the overview, login and sign-up routes.
 * For whatever reason the page would not react to any history changes and
 * would only every display the overview.
 */

describe('Auth', () => {
  let auth: AuthState;
  let preloadMock: jest.Mock;

  beforeAll(() => {
    config.disabled = true;
  });

  const testRender = () => {
    history.push('/auth');
    return _testRender(
      {
        auth,
        router: {
          location: {
            pathname: '/auth',
            state: undefined,
            search: '',
            query: {},
            hash: '',
          },
          action: 'PUSH'
        },
      },
      <Route path='/auth'>
        <Auth preloadMain={preloadMock} />
      </Route>
    );
  }

  beforeEach(() => {
    auth = {
      loading: false,
      signUpRequestSent: false,
      user: undefined,
    };

    preloadMock = jest.fn().mockResolvedValue(undefined);
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('is rendered correctly', () => {
    const {baseElement} = testRender();

    expect(baseElement).toMatchSnapshot();
  });
});