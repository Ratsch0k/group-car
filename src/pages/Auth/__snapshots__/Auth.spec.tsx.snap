// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth is rendered correctly 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root MuiBox-root makeStyles-root"
    >
      <div
        class="MuiBox-root MuiBox-root makeStyles-root"
      >
        <div
          class="spot-primary"
          id="spot1"
        />
        <div
          class="spot-primary-dark"
          id="spot2"
        />
        <div
          class="spot-secondary"
          id="spot3"
        />
      </div>
      <div
        class="MuiContainer-root makeStyles-contentRoot MuiContainer-maxWidthSm"
      >
        <div
          class="MuiPaper-root makeStyles-paperRoot MuiPaper-elevation15 MuiPaper-rounded makeStyles-paperRounded"
        >
          <div
            class="MuiBox-root MuiBox-root makeStyles-container"
          >
            <div
              class="MuiContainer-root makeStyles-root MuiContainer-maxWidthLg"
            >
              <svg
                height="100"
                version="1.1"
                viewBox="0 0 264.58 264.58"
                width="100"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="10"
                >
                  <path
                    d="m88.221 45h94.89"
                  />
                  <path
                    d="m110.52 31h21.776"
                  />
                  <path
                    d="m144.1 31 11.926 0.05635"
                  />
                  <path
                    d="m197.09 157 11.926 0.0563"
                  />
                  <path
                    d="m164.06 143.12h44.176"
                  />
                  <path
                    d="m139.74 143 11.926 0.0563"
                  />
                  <path
                    d="m181.03 199.07 11.926 0.0563"
                  />
                  <path
                    d="m187.35 185.06h21.776"
                  />
                  <path
                    d="m187.41 171h21.776"
                  />
                  <path
                    d="m182.43 59 11.926 0.05635"
                  />
                  <path
                    d="m182.89 73h21.776"
                  />
                  <path
                    d="m75.391 59h21.776"
                  />
                  <path
                    d="m60.117 73 11.926 0.05635"
                  />
                  <path
                    d="m72 87 11.926 0.05635"
                  />
                  <path
                    d="m192.66 87 11.926 0.05635"
                  />
                  <path
                    d="m58 101h21.776"
                  />
                  <path
                    d="m58 129 11.926 0.0564"
                  />
                  <path
                    d="m58 115h21.776"
                  />
                  <path
                    d="m58 143 11.926 0.0564"
                  />
                  <path
                    d="m72 157 11.926 0.0563"
                  />
                  <path
                    d="m62.153 171h21.776"
                  />
                  <path
                    d="m62.559 185.05 11.926 0.0563"
                  />
                  <path
                    d="m88.237 185 11.926 0.0563"
                  />
                  <path
                    d="m69.896 199.01h21.776"
                  />
                  <path
                    d="m79.718 213 11.926 0.0563"
                  />
                  <path
                    d="m105.57 213h21.776"
                  />
                  <path
                    d="m109.84 227h44.176"
                  />
                  <path
                    d="m168.09 227.14 11.926 0.0563"
                  />
                  <path
                    d="m154.93 213 11.926 0.0563"
                  />
                  <path
                    d="m180.82 213h21.776"
                  />
                  <g
                    fill="currentColor"
                    stroke="none"
                  >
                    <circle
                      cx="182.98"
                      cy="157"
                      r="5"
                    />
                    <circle
                      cx="58"
                      cy="157"
                      r="5"
                    />
                    <circle
                      cx="168.51"
                      cy="59"
                      r="5"
                    />
                    <circle
                      cx="86.307"
                      cy="73"
                      r="5"
                    />
                    <circle
                      cx="207.14"
                      cy="199"
                      r="5"
                    />
                    <circle
                      cx="84"
                      cy="143"
                      r="5"
                    />
                    <circle
                      cx="141.15"
                      cy="213"
                      r="5"
                    />
                    <circle
                      cx="84"
                      cy="129"
                      r="5"
                    />
                    <circle
                      cx="95.652"
                      cy="227"
                      r="5"
                    />
                    <circle
                      cx="105.66"
                      cy="199.01"
                      r="5"
                    />
                    <circle
                      cx="58"
                      cy="87"
                      r="5"
                    />
                  </g>
                </g>
              </svg>
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                <b>
                  My Group Car
                </b>
              </h5>
            </div>
            <div
              class="makeStyles-contentWrapper"
            >
              <div
                class="makeStyles-content"
              >
                <div
                  class="MuiContainer-root MuiContainer-maxWidthXs"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
                  >
                    <h6
                      class="MuiTypography-root MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                    >
                      auth.title
                    </h6>
                    <div
                      class="MuiGrid-root makeStyles-buttonContainer MuiGrid-item"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-contained MuiButton-containedPrimary MuiButton-fullWidth"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label makeStyles-buttonLabel"
                        >
                          form.login
                        </span>
                      </button>
                    </div>
                    <div
                      class="MuiGrid-root makeStyles-buttonContainer MuiGrid-item"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-contained MuiButton-containedPrimary MuiButton-fullWidth"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label makeStyles-buttonLabel"
                        >
                          form.sign-up
                        </span>
                      </button>
                    </div>
                    <p
                      class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                    >
                      auth.tryDemo
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root makeStyles-footer"
      >
        <div
          class="MuiGrid-root makeStyles-footerLinks MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorInherit"
            >
              privacyPolicy.title
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorInherit"
            >
              imprint.title
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
